sympy/geometry/point.py

```diff
@@ -442,32 +557,44 @@ def __add__(self, other):
 
         """
 
-        if iterable(other) and len(other) == len(self):
-            return Point([simplify(a + b) for a, b in zip(self, other)])
-        else:
-            raise ValueError(
-                "Points must have the same number of dimensions")
+        if iterable(other):
+            coords = [simplify(a + b) for a, b in zip(self, other)]
+            if len(coords) != self.ambient_dimension:
+                raise ValueError("Points must have the same number of dimensions")
+            if all(isinstance(c, Number) for c in coords):
```

Comment:
I don't know the history of `Point` turning all its arguments into exact types, so I didn't mess with that, but I tried to make it avoid coercing types when it wasn't needed.  (There are some bug's filed about poor numeric performance of `Point`).  I think the best behaviour would be for `Point(.1,.2) -> Point(.1,.2)`.  I suspect this would break a lot of the library though, since things like distance calculations, etc. would then be floats and I think a lot of the library has been programmed with the assumption of exactness...